modules/frontend/v1/frontend.go

```diff
@@ -203,43 +214,62 @@ func (f *Frontend) Process(server frontendv1pb.Frontend_ProcessServer) error {
 
 	lastUserIndex := queue.FirstUser()
 
+	reqBatch := &requestBatch{}
+	batchSize := 1
+	if querierSupportsBatching(querierFeatures) {
+		batchSize = f.cfg.MaxBatchSize
```

Comment:
Initially I wrote it so that the querier told the frontend a batch size. There are two places this can occur. 

1) In the initial TYPE_GET_ID request
2) In the response to every TYPE_HTTP_REQUEST*

I eventually backed off of it b/c even in the case of #2 the batch size will lag the actual state of the querier. Every GRPC streaming connection to the querier would have a possibly different batch size that would almost always be based on out of date information.

I like the ideal, but I'm concerned about the complexity and failure modes. I also considered an out of band streaming call that regularly updated the batch size between the querier <-> frontend globally for all streaming connections, but that seemed outside the scope of this PR.

In short, I think it's a good idea, but may take a bit to get there. I see this is a good first step in a more complete solution like that.